ssl/ssl_local.h

```diff
@@ -2209,6 +2209,40 @@ typedef enum downgrade_en {
 #define TLSEXT_SIGALG_ecdsa_brainpoolP384r1_sha384              0x081b
 #define TLSEXT_SIGALG_ecdsa_brainpoolP512r1_sha512              0x081c
 
+/* Sigalgs names */
+#define TLSEXT_SIGALG_ecdsa_secp256r1_sha256_name                    "ecdsa_secp256r1_sha256"
+#define TLSEXT_SIGALG_ecdsa_secp384r1_sha384_name                    "ecdsa_secp384r1_sha384"
+#define TLSEXT_SIGALG_ecdsa_secp521r1_sha512_name                    "ecdsa_secp521r1_sha512"
+#define TLSEXT_SIGALG_ecdsa_sha224_name                              "ecdsa_sha224"
+#define TLSEXT_SIGALG_ecdsa_sha1_name                                "ecdsa_sha1"
+#define TLSEXT_SIGALG_rsa_pss_rsae_sha256_name                       "rsa_pss_rsae_sha256"
+#define TLSEXT_SIGALG_rsa_pss_rsae_sha384_name                       "rsa_pss_rsae_sha384"
+#define TLSEXT_SIGALG_rsa_pss_rsae_sha512_name                       "rsa_pss_rsae_sha512"
+#define TLSEXT_SIGALG_rsa_pss_pss_sha256_name                        "rsa_pss_pss_sha256"
+#define TLSEXT_SIGALG_rsa_pss_pss_sha384_name                        "rsa_pss_pss_sha384"
+#define TLSEXT_SIGALG_rsa_pss_pss_sha512_name                        "rsa_pss_pss_sha512"
+#define TLSEXT_SIGALG_rsa_pkcs1_sha256_name                          "rsa_pkcs1_sha256"
+#define TLSEXT_SIGALG_rsa_pkcs1_sha384_name                          "rsa_pkcs1_sha384"
+#define TLSEXT_SIGALG_rsa_pkcs1_sha512_name                          "rsa_pkcs1_sha512"
+#define TLSEXT_SIGALG_rsa_pkcs1_sha224_name                          "rsa_pkcs1_sha224"
+#define TLSEXT_SIGALG_rsa_pkcs1_sha1_name                            "rsa_pkcs1_sha1"
+#define TLSEXT_SIGALG_dsa_sha256_name                                "dsa_sha256"
+#define TLSEXT_SIGALG_dsa_sha384_name                                "dsa_sha384"
+#define TLSEXT_SIGALG_dsa_sha512_name                                "dsa_sha512"
+#define TLSEXT_SIGALG_dsa_sha224_name                                "dsa_sha224"
+#define TLSEXT_SIGALG_dsa_sha1_name                                  "dsa_sha1"
+#define TLSEXT_SIGALG_gostr34102012_256_intrinsic_name               "gost2012_256"
+#define TLSEXT_SIGALG_gostr34102012_512_intrinsic_name               "gost2012_512"
+#define TLSEXT_SIGALG_gostr34102012_256_gostr34112012_256_name       "gost2012_256"
+#define TLSEXT_SIGALG_gostr34102012_512_gostr34112012_512_name       "gost2012_512"
```

Comment:
Would it make sense to distinguish `gostr34102012_*_intrinsic` from `gostr34102012_*_gostr34112012_*`? We currently use `gost2012_*` for all of them.

I also think  `TLSEXT_SIGALG_gostr34102012_256_gostr34112012_256_name` should be changed to `TLSEXT_SIGALG_gostr34102012_256_gostr34112012_name` (there is one more extra 256).
Same goes to `TLSEXT_SIGALG_gostr34102012_512_gostr34112012_512_name`.

thanks.

Reply:
> Would it make sense to distinguish gostr34102012_*_intrinsic from gostr34102012_*_gostr34112012_*? We currently use gost2012_* for all of them.

Conceptually sure -- but this decision to label all "Gost's" the same [pre-dates this PR](https://github.com/openssl/openssl/blob/fd39d1c80cd5bd9cb5c64e3fc96102397e5e860f/ssl/t1_trce.c#L594-L597), so touching this seems risky as this name may be externally used/referenced. Who could comment on this mainly Gost-related question? @beldmit?

> I also think TLSEXT_SIGALG_gostr34102012_256_gostr34112012_256_name should be changed to TLSEXT_SIGALG_gostr34102012_256_gostr34112012_name

Sure -- we can do this. But we could also use "TLSEXT_SIGALG_gostr34102012_gostr34112012_256_name" instead with the logic that the presumed strength (256) trails the presumed base algorithm name. Again, unsure what suits better the Gost spirit (sorry, couldn't resist the pun :).

Oh, and this naming again has nothing really to do with this PR but is just following what is in the current "master":

https://github.com/openssl/openssl/blob/fd39d1c80cd5bd9cb5c64e3fc96102397e5e860f/ssl/t1_trce.c#L594-L597